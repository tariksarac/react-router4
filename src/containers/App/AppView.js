import React from 'react'
import { Switch, Route, Redirect } from 'react-router-dom'

import HeaderContainer from '../Header/HeaderContainer'
import HomeContainer from '../Home/HomeContainer'
import NotFoundPage from '../../components/NotFoundPage/NotFoundPage'
import Authentication from '../../containers/Authentication/AuthenticationContainer'
import Register from '../../components/RegisterPage/RegisterPage'


//High order component for handling protected routes
const PrivateRoute = ({ component: Component, ...rest }) => (
    <Route {...rest} render={props => (
        rest.auth ? (
            <Component {...props}/>
        ) : (
            <Redirect to={{
                pathname: '/login',
                state: { from: props.location }
            }}/>
        )
    )}/>
);

class App extends React.Component {

    render() {
        return(
            <div>
                <HeaderContainer />
                <Switch>
                    <Route exact path="/" component={Register}/>
                    <Route path="/auth" component={Authentication}/>
                    <PrivateRoute path="/protected" component={HomeContainer} auth={this.props.authenticated}/>
                    <Route component={NotFoundPage}/>
                </Switch>
            </div>
        )
    }
}

export default App


