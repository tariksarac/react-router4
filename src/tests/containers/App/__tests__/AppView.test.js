import React from 'react'
import AppView from '../../../../containers/App/AppView'
import { shallow } from 'enzyme'


it('should render App without crashing', () => {
    const app = shallow(<AppView />);
    expect(app).toMatchSnapshot()
});