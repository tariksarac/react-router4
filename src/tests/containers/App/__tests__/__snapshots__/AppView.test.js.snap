// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App without crashing 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div>
    <withRouter(Connect(Header)) />
    <Switch>
        <Route
            component={[Function]}
            exact={true}
            path="/"
        />
        <Route
            component={[Function]}
            path="/auth"
        />
        <PrivateRoute
            auth={undefined}
            component={[Function]}
            path="/protected"
        />
        <Route
            component={[Function]}
        />
    </Switch>
</div>,
  "nodes": Array [
    <div>
      <withRouter(Connect(Header)) />
      <Switch>
            <Route
                  component={[Function]}
                  exact={true}
                  path="/"
            />
            <Route
                  component={[Function]}
                  path="/auth"
            />
            <PrivateRoute
                  auth={undefined}
                  component={[Function]}
                  path="/protected"
            />
            <Route
                  component={[Function]}
            />
      </Switch>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <App />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": App {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div>
          <withRouter(Connect(Header)) />
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              path="/auth"
                    />
                    <PrivateRoute
                              auth={undefined}
                              component={[Function]}
                              path="/protected"
                    />
                    <Route
                              component={[Function]}
                    />
          </Switch>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div>
          <withRouter(Connect(Header)) />
          <Switch>
                    <Route
                              component={[Function]}
                              exact={true}
                              path="/"
                    />
                    <Route
                              component={[Function]}
                              path="/auth"
                    />
                    <PrivateRoute
                              auth={undefined}
                              component={[Function]}
                              path="/protected"
                    />
                    <Route
                              component={[Function]}
                    />
          </Switch>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <App />,
}
`;
